### PR TITLE
Add checkstyle rule to show final local variable violations as warning

### PR DESCRIPTION
**Pull request by XiangRongLin**
#### What is it?
- [x] Code base improvement (dev facing)

#### Description of the changes in your PR

Add checkstyle rule to check for final local variable violations. 
Since fixing everything at once would be very bothersome, it is shown as warning and not error. Instead it can be gradually fixed when someone touches the files.
The reason i want it is, that on every PR one has to manually search for the violations.

Doc: https://checkstyle.sourceforge.io/apidocs/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.html

#### Agreement
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.

### CODE REVIEW

File: checkstyle.xml [hunk 1]
```diff
@@ -154,6 +154,11 @@
         <module name="MultipleVariableDeclarations"/>
         <module name="SimplifyBooleanExpression"/>
         <module name="SimplifyBooleanReturn"/>
+        <module name="FinalLocalVariable">
+            <property name="tokens" value="VARIABLE_DEF,PARAMETER_DEF"/>
```
Comment (wb9688) [inline checkstyle.xml:158]: Remove this line, it's pointless, as we require all parameters to be final with that other rule anyway.

Comment (wb9688) [reply checkstyle.xml:158]: Hmm… it seems to make a significant difference in amount of warnings

**Comment by TobiGr** [comment]
Happy rebasing to everyone!?

**Comment by wb9688** [comment]
@TobiGr: I guess so… ¯\_(ツ)_/¯

**Pull request merged**